honeybee/model.py

```diff
@@ -119,243 +207,391 @@ def north_vector(self, value):
-        for door in other_model.doors:
-            if not door.has_parent:
-                self.add_door(door)
+        for room in other_model._rooms:
+            self._rooms.append(room)
+        for face in other_model._faces:
+            self._faces.append(face)
+        for shade in other_model._shades:
+            self._shades.append(shade)
+        for aperture in other_model._apertures:
+            self._apertures.append(aperture)
+        for door in other_model._doors:
+            self._doors.append(door)
 
     def add_room(self, obj):
         """Add a Room object to the model."""
         assert isinstance(obj, Room), 'Expected Room. Got {}.'.format(type(obj))
         self._rooms.append(obj)
-        self._room_names.append(obj.name)
-        for face in obj.faces:
-            self.add_face(face)
-        for shade in obj.indoor_shades:
-            self.add_shade(shade)
-        for shade in obj.outdoor_shades:
-            self.add_shade(shade)
 
     def add_face(self, obj):
-        """Add a Face object to the model."""
+        """Add a Face object without a parent to the model."""
         assert isinstance(obj, Face), 'Expected Face. Got {}.'.format(type(obj))
+        assert not obj.has_parent, 'Face "{}"" has a parent Room. Add the Room to '\
+            'the model instead of the Face.'.format(obj.name)
         self._faces.append(obj)
-        self._face_names.append(obj.name)
-        for ap in obj.apertures:
-            self.add_aperture(ap)
-        for dr in obj.doors:
-            self.add_door(dr)
 
     def add_shade(self, obj):
         """Add an Shade object to the model."""
         assert isinstance(obj, Shade), 'Expected Shade. Got {}.'.format(type(obj))
+        assert not obj.has_parent, 'Shade "{}"" has a parent Room. Add the Room to '\
+            'the model instead of the Shade.'.format(obj.name)
         self._shades.append(obj)
-        self._shade_names.append(obj.name)
 
     def add_aperture(self, obj):
         """Add an Aperture object to the model."""
         assert isinstance(obj, Aperture), 'Expected Aperture. Got {}.'.format(type(obj))
+        assert not obj.has_parent, 'Aperture "{}"" has a parent Face. Add the Face to '\
+            'the model instead of the Aperture.'.format(obj.name)
         self._apertures.append(obj)
-        self._aperture_names.append(obj.name)
 
     def add_door(self, obj):
         """Add an Door object to the model."""
         assert isinstance(obj, Door), 'Expected Door. Got {}.'.format(type(obj))
+        assert not obj.has_parent, 'Door "{}"" has a parent Face. Add the Face to '\
+            'the model instead of the Door.'.format(obj.name)
         self._doors.append(obj)
-        self._door_names.append(obj.name)
+
+    def get_rooms_by_name(self, names):
+        """Get a list of Room objects in the model given the Room names."""
+        rooms = []
+        model_rooms = self._rooms
+        for name in names:
```

Comment:
Not many people use it but Python supports `else` for `for` loops which can simplify cases like this. You can write this loop as:

```
        for name in names:
            for room in model_rooms:
                if room.name == name:
                    rooms.append(room)
                    break
            else:
                raise ValueError('Room "{}" was not found in the model.'.format(name))
```

Reply:
Very good to know.  Thank you for the tip and I will implement it.

Reply:
Now that you are at it you may want to update this one too:

https://github.com/ladybug-tools/honeybee-core/blob/master/honeybee/_lockable.py#L85-L94

Something like this should work:

```python
    if hasattr(cls, '__slots__'):
        for parent_class in getmro(cls)[:-1]:
            if '_locked' in parent_class.__slots__:
                break
        else:
            raise AttributeError(
                'When using the @locakble decorator on a class with __slots__, '
                'a "_locked" variable must be specified within __slots__.')
```